### PR TITLE
Add git_shallow_fetch_since to checkout.py

### PR DESCRIPTION
**Pull request by elsaferrara**

### CODE REVIEW

**Comment by leocardao** [comment]
To correct the failure of the pre-commit CI, simply rebase (I've just merged a correction for this)

File: src/e3/anod/checkout.py [hunk 1]
```diff
@@ -243,15 +243,26 @@ def update_git(
...
                 "E3_ENABLE_FEATURE", ""
-            ).split(",") and (not self.compute_changelog or not old_commit)
+            ).split(",")
+
+            if shallow_fetch and (not self.compute_changelog or not old_commit):
+                shallow_cmd = "--depth=1"
+
+            for feature in os.environ.get("E3_ENABLE_FEATURE", "").split(","):
+                if "git_fetch_shallow_since" in feature:
+                    date = feature.replace("git_fetch_shallow_since=", "")
```
Comment (pierretr) [inline src/e3/anod/checkout.py:258]:
```suggestion
                    date = feature.removeprefix("git_fetch_shallow_since=")
```

File: src/e3/anod/checkout.py [hunk 2]
```diff
@@ -243,15 +243,26 @@ def update_git(
...
+                    shallow_cmd = f"--shallow-since={date}"
```
Comment (pierretr) [inline src/e3/anod/checkout.py:259]: We do not want to override `--depth=1` if set.

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Version 22.7.0 (2024-??-??) *NOT RELEASED YET*
 
+* Add git_shallow_fetch_since to checkout.py
```
Comment (grouigrokon) [inline NEWS.md:3]: Thanks !

**Review by pierretr — APPROVED** [review]

**Pull request merged**